### PR TITLE
Travis CI

### PR DESCRIPTION
**Pull request by avilchess**
Make sure you have checked _all_ steps below.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Benchmarks
- [x] My PR adds the following micro benchmarks __OR__ does not need benchmarks for this extremely good reason:

### Commits
- [x] My commits have been squashed if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

## License
- [x] Add a [Mozilla Public License 2.0 license header](http://mozilla.org/MPL/2.0/) to all the new files.

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.

### CODE REVIEW

File: .CI/travis/install-arrayfire_linux.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# Copyright (c) 2018 Shapelets.io
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+wget http://arrayfire.s3.amazonaws.com/3.6.0/ArrayFire-no-gl-v3.6.0_Linux_x86_84.sh
```
Comment (otorreno) [inline .CI/travis/install-arrayfire_linux.sh]: We should work with AF v3.5.1

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,6 @@ build
 CMakeCache.txt
 include/khiva/version.h
 test/versionTest.cpp
+.idea/
```
Comment (otorreno) [inline .gitignore]: do we need it?

File: CMakeLists.txt [hunk 3]
```diff
@@ -30,6 +30,21 @@ OPTION(KHIVA_BUILD_BENCHMARKS "Build benchmarks of the Khiva library" ON)
 OPTION(KHIVA_BUILD_EXAMPLES "Build examples of the Khiva library" ON)
 OPTION(KHIVA_BUILD_DOCUMENTATION "Create and install the HTML based API documentation (requires Doxygen, GraphViz and Sphinx)" ON)
 OPTION(KHIVA_USE_CONAN "Use the conan package manager to download the dependencies of the Khiva library" ON)
+OPTION(KHIVA_ENABLE_COVERAGE "This flag allows Code Coverage of the code" OFF)
+OPTION(KHIVA_ONLY_CPU_BACKEND "This flag only allows the execution on the CPU Backend of the code" OFF)
```
Comment (otorreno) [inline CMakeLists.txt]: This flag **limits** the ...

File: CMakeLists.txt [hunk 4]
```diff
@@ -30,6 +30,21 @@ OPTION(KHIVA_BUILD_BENCHMARKS "Build benchmarks of the Khiva library" ON)
 OPTION(KHIVA_BUILD_EXAMPLES "Build examples of the Khiva library" ON)
 OPTION(KHIVA_BUILD_DOCUMENTATION "Create and install the HTML based API documentation (requires Doxygen, GraphViz and Sphinx)" ON)
 OPTION(KHIVA_USE_CONAN "Use the conan package manager to download the dependencies of the Khiva library" ON)
+OPTION(KHIVA_ENABLE_COVERAGE "This flag allows Code Coverage of the code" OFF)
```
Comment (otorreno) [inline CMakeLists.txt]: This flag **enables** Code ...

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -37,11 +37,17 @@ TARGET_INCLUDE_DIRECTORIES(${KHIVALIB} PUBLIC
...
                            ${ArrayFire_INCLUDE_DIRS}
                            ${Boost_INCLUDE_DIRS})
-
-# Declare compilation libraries
-TARGET_LINK_LIBRARIES(${KHIVALIB}
-                      ${ArrayFire_Unified_LIBRARIES}
+IF(KHIVA_ONLY_CPU_BACKEND)
+    # Declare compilation libraries
+    TARGET_LINK_LIBRARIES(${KHIVALIB}
+                      ${ArrayFire_CPU_LIBRARIES}
```
Comment (raulbocanegra) [inline src/CMakeLists.txt]: Do you think that using CMake targets is a better approach than using legacy variables taking into account that the projcet reuqires a modern CMake version.
From the AF lib:

>  ``ArrayFire::af``
   Target for the ArrayFire Unified backend.
 ``ArrayFire::afcpu``
   Target for the ArrayFire CPU backend.
 ``ArrayFire::afcuda``
   Target for the ArrayFire CUDA backend.
 ``ArrayFire::afopencl``
   Target for the ArrayFire OpenCL backend.

**Pull request merged**